source/extensions/filters/udp/dns_filter/dns_filter.cc

```diff
@@ -174,10 +227,15 @@ DnsLookupResponseCode DnsFilter::getResponseForQuery(DnsQueryContextPtr& context
         continue;
       }
     }
-    // TODO(abaptiste): resolve the query externally
+
+    ENVOY_LOG(debug, "resolving name [{}] via external resolvers", query->name_);
```

Comment:
Keep in mind that C-Ares will return an error if there is more than one query in the buffer.   

So, yes, in this case since the external resolution path is asynchronous, this has to be done.  We do not have an answer for the client and have to "return" so that we do not block the listener while waiting for the DNS response.  When we get the response from the external resolver, we construct the response and send it.

In the case where there is a mix of locally resolved queries and externally resolved queries, we could send 2 separate DNS answers back to the client.  This ties back to C-Ares treating multiple queries as an error.  I haven't found a tool yet where this is supported.

Reply:
No, I agree this cannot be handled now. It will be good to capture this in the code and return an error if there are more than 1 queries in the external query case and mixed queries case.

Reply:
Cool.  I will update this in the code and docs.